### PR TITLE
Govt StateID and DistrictId changes

### PR DESCRIPTION
**Pull request by ravishanigarapu**
## 📋 Description

JIRA ID: AMM-1166

Please provide a summary of the change and the motivation behind it. Include relevant context and details.

---

## ✅ Type of Change

- [ ] 🐞 **Bug fix** (non-breaking change which resolves an issue)
- [ ] ✨ **New feature** (non-breaking change which adds functionality)
- [ ] 🔥 **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🛠 **Refactor** (change that is neither a fix nor a new feature)
- [ ] ⚙️ **Config change** (configuration file or build script updates)
- [ ] 📚 **Documentation** (updates to docs or readme)
- [ ] 🧪 **Tests** (adding new or updating existing tests)
- [ ] 🎨 **UI/UX** (changes that affect the user interface)
- [ ] 🚀 **Performance** (improves performance)
- [ ] 🧹 **Chore** (miscellaneous changes that don't modify src or test files)

---

## ℹ️ Additional Information

Please describe how the changes were tested, and include any relevant screenshots, logs, or other information that provides additional context.

## Summary by CodeRabbit

- **New Features**
	- Added new properties for government state and district IDs in the Districts and States classes.
	- Enhanced constructors in Districts and States to accept additional parameters.

- **Bug Fixes**
	- Improved data retrieval in the DistrictMasterRepo and StateMasterRepo to include new fields in the queries.

- **Chores**
	- Updated validation logic in the HTTPRequestInterceptor to enhance request processing and error handling.

### CODE REVIEW

File: src/main/java/com/iemr/tm/data/location/States.java [hunk 1]
```diff
@@ -43,6 +43,10 @@ public class States {
 	@Column(name = "StateID")
 	@Expose
 	private Integer stateID;
+	
+	@Column(name = "GovtStateID")
+	@Expose
+	private Integer govtLGDStateID;
```
[the remark this answers is not in the document]
Comment (ravishanigarapu) [reply src/main/java/com/iemr/tm/data/location/States.java:49]: @coderabbitai In DB its mentioned same

File: src/main/java/com/iemr/tm/data/location/States.java [hunk 2]
```diff
@@ -155,5 +159,10 @@
 	public void setStateName(String stateName) {
 		this.stateName = stateName;
 	}
-
+	
+	public States(int StateID, String StateName, Integer govtLGDStateID) {
+		this.stateID = StateID;
+		this.stateName = StateName;
+		this.govtLGDStateID = govtLGDStateID;
+	}
```
[the remark this answers is not in the document]
Comment (ravishanigarapu) [reply src/main/java/com/iemr/tm/data/location/States.java:167]: @coderabbitai it is used for other methods its not deprecated

File: src/main/java/com/iemr/tm/utils/http/HTTPRequestInterceptor.java [hunk 3]
```diff
@@ -86,6 +90,11 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
 					status = false;
 					break;
 				default:
+					String remoteAddress = request.getHeader("X-FORWARDED-FOR");
+					if (remoteAddress == null || remoteAddress.trim().length() == 0) {
+						remoteAddress = request.getRemoteAddr();
+					}
+					validator.checkKeyExists(authorization, remoteAddress);
```
[the remark this answers is not in the document]
Comment (ravishanigarapu) [reply src/main/java/com/iemr/tm/utils/http/HTTPRequestInterceptor.java:97]: @coderabbitai its mandatory thing

File: src/main/java/com/iemr/tm/data/location/Districts.java [hunk 4]
```diff
@@ -86,6 +86,13 @@ public Integer getStateID() {
 	public void setStateID(Integer stateID) {
 		this.stateID = stateID;
 	}
+	@Column(name = "GovtStateID")
+	@Expose
+	private Integer govtLGDStateID;
+	
+	@Column(name = "GovtDistrictID")
+	@Expose
+	private Integer govtLGDDistrictID;
```
[the remark this answers is not in the document]
Comment (ravishanigarapu) [reply src/main/java/com/iemr/tm/data/location/Districts.java:95]: @coderabbitai using constaructor not required

**Review by helenKaryamsetty — APPROVED** [review]

**Pull request merged**